src/main/java/org/avarion/graves/Graves.java

```diff
@@ -194,6 +249,18 @@ private void registerMetrics() {
     }
 
     public void registerListeners() {
+        // Configurable death listener priority
+        String priorityStr = getConfig().getString("settings.listener-priority.death");
```

Comment:
Maybe make an extra method for this part. So the `registerListeners` method stays clean (only `registerEvents` calls).
```java
private EventPriority getDeathListenerPriority() {
        String priorityStr = getConfig().getString("settings.listener-priority.death");

        try {
            return EventPriority.valueOf(priorityStr.toUpperCase());
        } catch (IllegalArgumentException e) {
            getLogger().warning("Invalid event priority in config: " + priorityStr + ". Defaulting to HIGHEST.");
            return EventPriority.HIGHEST;
        }
```